### PR TITLE
PRIME-2346 Update HA site registration step from 8 steps to 5 steps

### PR DESCRIPTION
**Pull request by bergomi02**

### CODE REVIEW

**Comment by neophyte57** [comment]
When I attempt "Save and Continue", web app doesn't let me advance without address (correct) but doesn't indicate the reason (incorrect)

image

**Comment by bergomi02** [comment]
> When I attempt "Save and Continue", web app doesn't let me advance without address (correct) but doesn't indicate the reason (incorrect)
> 
> image

this is fixed.

**Comment by neophyte57** [comment]
First one mentioned this morning:
1. Add new site
2. Choose a value for each of Care type and Vendor
3. Click "Back" button
4. Choose "Discard changes"

image

**Comment by neophyte57** [comment]
Second one mentioned this morning:
1. Enter something for Site Name
2. Click "Back" button
3. Choose "Discard changes"
4. Click "Save and Continue" button
5. Click "Back" button
6. Get the following, but incorrect since there should be nothing to Discard 

image

**Comment by bergomi02** [comment]
> First one mentioned this morning:
> 
> 1. Add new site
> 2. Choose a value for each of Care type and Vendor
> 3. Click "Back" button
> 4. Choose "Discard changes"
> 
> image

This should be fixed. However, I discovery a new issue where operation hours will be saved even using back button. I will address this later.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg-routing.module.ts [hunk 1]
```diff
@@ -112,13 +103,7 @@ const routes: Routes = [
...
             component: HealthAuthCareTypePageComponent,
             canDeactivate: [CanDeactivateFormGuard],
-            data: { title: 'Health Authority Care Type' }
-          },
-          {
-            path: HealthAuthSiteRegRoutes.SITE_ADDRESS,
-            component: SiteAddressPageComponent,
-            canDeactivate: [CanDeactivateFormGuard],
-            data: { title: 'Site Address' }
+            data: { title: 'Health Authority Care Type & Vendor' }
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg-routing.module.ts:106]: Mock-up splits into two sub-titles

image

Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg-routing.module.ts]: this is fixed.

**Comment by neophyte57** [comment]
I don't have exact reproduction steps but it starts around "Hours of Operation" surprisingly, while saving changes on that page.  It doesn't happen with every update so it's hard to pinpoint what causes the `duplicate key value violates unique constraint "PK_Address"` to begin occurring 

image

**Comment by neophyte57** [comment]
This seems to be an existing bug reproducible in `dev`.  What AU sees:

image

What Admin sees:
image

**Comment by neophyte57** [comment]
> > 
> 
> The time display issue should be fixed. However, I can't reproduce the server error on the operation hours page and I can't fix that.

No problem, possibly it was fixed indirectly when you addressed "Hours of Operation - does not warn re: Discard Changes and also actually saves data entered (unexpected to user)" issue.

**Comment by neophyte57** [comment]
Can you check whether this is correct or should it be "site"?  

image

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/services/health-authority-site-form-state.service.ts [hunk 2]
```diff
@@ -146,14 +128,29 @@ export class HealthAuthoritySiteFormStateService extends AbstractFormStateServic
...
     this.healthAuthCareTypeFormState.patchValue(healthAuthoritySite);
-    this.siteAddressFormState.patchValue(healthAuthoritySite);
     this.hoursOperationFormState.patchValue(healthAuthoritySite);
     this.administratorFormState.patchValue(healthAuthoritySite);
-    this.technicalSupportFormState.patchValue(healthAuthoritySite);
+
+    this.markAsPristine();
+  }
+
+  public patchSiteInfomrationForm(healthAuthoritySite: HealthAuthoritySite) {
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/services/health-authority-site-form-state.service.ts:139]: Typo in `patchSiteInfomrationForm`

File: prime-angular-frontend/src/app/shared/components/address-view/address-view.component.ts [hunk 3]
```diff
@@ -29,6 +29,11 @@ export class AddressViewComponent implements OnInit {
    * Show the redirect icon.
    */
   @Input() public showRedirect: boolean;
+  /**
+   * @description
+   * Show the edit button at the bottom
+   */
+  @Input() public showEditRedirect: boolean;
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/address-view/address-view.component.ts:36]: Both `showEditRedirect` and `showRedirect` enable editing so should `showEditRedirect` be named `showEditButton` instead?

File: prime-angular-frontend/src/app/shared/components/address-view/address-view.component.html [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
   <ng-container *ngTemplateOutlet="(hasAddress) ? addressIf : addressElse"></ng-container>
 
-  <button *ngIf="showRedirect"
+  <button *ngIf="showEditRedirect"
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/address-view/address-view.component.html:17]: Wait, I'm confused about this change and why it's necessary.  Might need to chat about it

**Comment by bergomi02** [comment]
> Can you check whether this is correct or should it be "site"? If it should be "site", can address in other PR #2230
> 
> image

According to the mock up, it is correct. 
image
Probably confirm with client in standup quickly as well.

File: prime-angular-frontend/src/app/shared/components/address-view/address-view.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/address-view/address-view.component.ts:36]: This change is not specific for this ticket. It fixes an existing bug where each component has two redirect controls: an icon and a button. For Address component, it was missing a  boolean variable for the button.

File: prime-angular-frontend/src/app/shared/components/address-view/address-view.component.html (hunk 4 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/address-view/address-view.component.html:17]: In there, it just links the new variable to the button.

**Review by neophyte57 — APPROVED** [review]
Great work

**Pull request merged**